src/Abp/Runtime/Caching/AbpCacheBase.cs

```diff
@@ -258,21 +259,52 @@ public virtual async Task<TValue[]> GetAsync(TKey[] keys, Func<TKey, Task<TValue
             return items;
         }
 
-        public abstract TValue GetOrDefault(TKey key);
+        protected abstract ConditionalValue<TValue> TryGetValue(TKey key);
```

Comment:
wouldn't keeping the same method signature is better? given that this method is protected only.

Reply:
I think a common and less verbose usage is better.

We probably want to make these methods public (and deprecate `GetOrDefault`) once stable.

Reply:
@acjh I couldn’t think of a better reason to make TryGetValue (as a way to detect cache miss) as public methods unless we are thinking AbpCache as a dictionary.

Reply:
*A way to detect cache miss* is precisely what a cache should provide.
Both [`IDistributedCache`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.distributed.idistributedcache?view=dotnet-plat-ext-3.1) and [`IMemoryCache`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.imemorycache?view=dotnet-plat-ext-3.1) do.

`GetOrDefault` and even `Get` with factory are helper methods rather than core methods of a cache.

Reply:
ok sounds reasonable to expose as public API then 😄